tetrad-lib/src/main/java/edu/cmu/tetrad/data/CovariancesFloatStandard.java

```diff
@@ -0,0 +1,160 @@
+                covariances[i][j] = (float) matrix[i][j];
+            }
+        }
+
+        this.numOfCols = matrix.length;
+        this.numOfRows = sampleSize;
+        this._data = null;
+    }
+
+    public float[] computeLowerTriangle(boolean biasCorrected) {
+        float[] covarianceMatrix = new float[(numOfCols * (numOfCols + 1)) / 2];
+
+        center();
+
+        int index = 0;
+        for (int col = 0; col < numOfCols; col++) {
+            for (int col2 = 0; col2 < col; col2++) {
+                float variance = 0;
+                for (int row = 0; row < numOfRows; row++) {
+                    variance += ((_data[row][col]) * (_data[row][col2]));
+                }
+                variance /= numOfRows - 1;
+                covarianceMatrix[index++] = biasCorrected ? variance * ((float) numOfRows / (float) (numOfRows - 1)) : variance;
+            }
+            float variance = 0;
+            for (int row = 0; row < numOfRows; row++) {
+                variance += ((_data[row][col]) * (_data[row][col]));
+            }
+            variance /= numOfRows - 1;
+            covarianceMatrix[index++] = biasCorrected ? variance * ((float) numOfRows / (float) (numOfRows - 1)) : variance;
+        }
+
+        return covarianceMatrix;
+    }
+
+    public float[][] compute(boolean biasCorrected) {
+        float[][] covarianceMatrix = new float[numOfCols][numOfCols];
+
+        center();
+
+        for (int col1 = 0; col1 < numOfCols; col1++) {
+            for (int col2 = 0; col2 < col1; col2++) {
+                float cov = 0;
+                for (int row = 0; row < numOfRows; row++) {
+                    cov += ((_data[row][col1]) * (_data[row][col2]));
+                }
+                cov /= numOfRows - 1;
+                cov = biasCorrected ? cov * ((float) numOfRows / (float) (numOfRows - 1)) : cov;
+                covarianceMatrix[col1][col2] = cov;
+                covarianceMatrix[col2][col1] = cov;
+            }
+            float variance = 0;
+            for (int row = 0; row < numOfRows; row++) {
+                variance += ((_data[row][col1]) * (_data[row][col1])) / (numOfRows - 1);
+            }
+            covarianceMatrix[col1][col1] = biasCorrected ? variance * ((float) numOfRows / (float) (numOfRows - 1)) : variance;
+        }
+
+        return covarianceMatrix;
+    }
+
+    private void center() {
+        for (int col = 0; col < numOfCols; col++) {
+            double mean = 0;
+            for (int row = 0; row < numOfRows; row++) {
+                mean += _data[row][col];
+            }
+            mean /= numOfRows;
+            for (int row = 0; row < numOfRows; row++) {
+                _data[row][col] -= mean;
+            }
+        }
+    }
+
+    @Override
+    public double covariance(int i, int j) {
+        return covariances[i][j];
+    }
+
+    public int size() {
+        return numOfCols;
+    }
+
+    public void setCovariance(int i, int j, double v) {
```

Comment:
This method should not exist.  A user should not be able to modify the covariance matrix.  This will lead to data corruption.